### ISSUE TITLE
Provide K8S error output

### ISSUE THREAD

**Issue by timbuchwaldt** [bundlewrap/bundlewrap#397]
Just got started using BW for K8S setup and ran into the first issue when I introduced a typo in a manifest and got no output except "error".
It would be great to get the output of the underlying `kubectl` command.

**Comment by trehn** [comment]
This isn't really specific to Kubernetes. The output you're looking for is available with `bw --debug`. I've traditionally been hesitant to just dump the output of other commands into the output of `bw apply` because I didn't want it to get polluted with Java stacktraces and other crap like that, rendering the output of a large `bw apply` completely unreadable.

The current situation of just marking the item as "failed" without any further explanation should definitely be improved. [BW_DEBUG_LOG_DIR](http://docs.bundlewrap.org/guide/env/#bw_debug_log_dir) was created as an easy first fix. Maybe it would be better to dump these errors (and just the errors) into a temporary log file if and when they occur and then just reference the path to that file in `bw apply` output?

**Comment by janpieper** [comment]
Here is an example with a failing `bw apply` that gives you NO information about what is failing... Debug output is enabled.

<sub>Hint: `spec.spec.template.containers[0].ports[0].name` in `deployment.yaml` is empty</sub>

<hr />

**items.py**

```python
k8s_namespaces = {"br0ken": {}}

k8s_deployments = {
    "br0ken/nginx": {
        "manifest_file": "deployment.yaml",
        "manifest_processor": "jinja2",
        "context": {},
    }
}

k8s_services = {
    "br0ken/nginx": {
        "manifest_file": "service.yaml",
        "manifest_processor": "jinja2",
        "context": {},
    }
}
```

**service.yaml**

```yaml
spec:
  type: NodePort
  selector:
    app: nginx
  ports:
    - protocol: TCP
      port: 80
      targetPort: 80
```

**deployment.yaml**

```yaml
spec:
  replicas: 1
  template:
    metadata:
      name: nginx
      labels:
        app: nginx
    spec:
      containers:
        - name: nginx
          image: nginx:1.15.8
          imagePullPolicy: Always
          ports:
            - name:
              containerPort: 80
              protocol: TCP
```

```
$ bw -d apply br0ken
[2019-02-06 14:26:07.363267] [DEBUG] invocation: /home/jan/.local/bin/bw -d apply br0ken
[2019-02-06 14:26:07.452589] [DEBUG] node br0ken gets its dummy attribute from: default
[2019-02-06 14:26:07.453698] [DEBUG] spinning up worker pool apply
[2019-02-06 14:26:07.453762] [DEBUG] worker pool apply is starting task br0ken on worker #0
[2019-02-06 14:26:07.454020] i br0ken  started at 2019-02-06 14:26:07
[2019-02-06 14:26:07.454088] [DEBUG] node br0ken gets its os attribute from: node
[2019-02-06 14:26:07.454159] [DEBUG] node br0ken gets its os_version attribute from: default
[2019-02-06 14:26:07.454222] [DEBUG] worker pool apply waiting for next task to complete
[2019-02-06 14:26:07.454626] [DEBUG] spinning up worker pool apply_br0ken
[2019-02-06 14:26:07.454700] [DEBUG] worker pool apply_br0ken is starting task br0ken:k8s_namespace:br0ken on worker #0
[2019-02-06 14:26:07.454982] [DEBUG] worker pool apply_br0ken waiting for next task to complete
[2019-02-06 14:26:07.455102] [DEBUG] node br0ken gets its kubectl_context attribute from: node
[2019-02-06 14:26:07.455392] [DEBUG] running command with ID KCIJ: kubectl --context=home --namespace=br0ken get -o json Namespace br0ken
[2019-02-06 14:26:07.759292] [DEBUG] command with ID KCIJ finished with return code 0
[2019-02-06 14:26:07.761164] [DEBUG] worker pool apply_br0ken delivering result of br0ken:k8s_namespace:br0ken on worker #0
[2019-02-06 14:26:07.761585] [DEBUG] worker pool apply_br0ken is starting task br0ken:k8s_namespace: on worker #1
[2019-02-06 14:26:07.762597] [DEBUG] worker pool apply_br0ken is starting task br0ken:k8s_service:br0ken/nginx on worker #2
[2019-02-06 14:26:07.763626] [DEBUG] worker pool apply_br0ken is starting task br0ken:k8s_deployment:br0ken/nginx on worker #3
[2019-02-06 14:26:07.771161] [DEBUG] br0ken:br0ken:k8s_service:br0ken/nginx: rendering with Jinja2...
[2019-02-06 14:26:07.771388] [DEBUG] worker pool apply_br0ken waiting for next task to complete
[2019-02-06 14:26:07.773120] [DEBUG] worker pool apply_br0ken delivering result of br0ken:k8s_namespace: on worker #1
[2019-02-06 14:26:07.772910] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendering with Jinja2...
[2019-02-06 14:26:07.773460] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendered in 4e-05s
[2019-02-06 14:26:07.773535] [DEBUG] worker pool apply_br0ken waiting for next task to complete
[2019-02-06 14:26:07.771879] [DEBUG] br0ken:br0ken:k8s_service:br0ken/nginx: rendered in 8.8e-05s
[2019-02-06 14:26:07.777655] [DEBUG] running command with ID 5H0S: kubectl --context=home --namespace=br0ken get -o json Deployment nginx
[2019-02-06 14:26:07.780381] [DEBUG] running command with ID MLOE: kubectl --context=home --namespace=br0ken get -o json Service nginx
[2019-02-06 14:26:08.186599] [DEBUG] command with ID MLOE finished with return code 0
[2019-02-06 14:26:08.187020] [DEBUG] command with ID 5H0S finished with return code 0
[2019-02-06 14:26:08.191276] [DEBUG] br0ken:br0ken:k8s_service:br0ken/nginx: rendering with Jinja2...
[2019-02-06 14:26:08.193690] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendering with Jinja2...
[2019-02-06 14:26:08.194177] [DEBUG] br0ken:br0ken:k8s_service:br0ken/nginx: rendered in 0.000141s
[2019-02-06 14:26:08.194710] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendered in 9.2e-05s
[2019-02-06 14:26:08.212752] [DEBUG] worker pool apply_br0ken delivering result of br0ken:k8s_service:br0ken/nginx on worker #2
[2019-02-06 14:26:08.213968] [DEBUG] worker pool apply_br0ken is starting task br0ken:k8s_service: on worker #0
[2019-02-06 14:26:08.214668] [DEBUG] worker pool apply_br0ken waiting for next task to complete
[2019-02-06 14:26:08.217842] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendering with Jinja2...
[2019-02-06 14:26:08.218469] [DEBUG] worker pool apply_br0ken delivering result of br0ken:k8s_service: on worker #0
[2019-02-06 14:26:08.218997] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendered in 0.000103s
[2019-02-06 14:26:08.226274] [DEBUG] worker pool apply_br0ken waiting for next task to complete
[2019-02-06 14:26:08.234874] [DEBUG] running command with ID MHHY: kubectl --context=home --namespace=br0ken apply -f -
[2019-02-06 14:26:08.844831] [DEBUG] command with ID MHHY finished with return code 0
[2019-02-06 14:26:08.846208] [DEBUG] running command with ID VA1V: kubectl --context=home --namespace=br0ken get -o json Deployment nginx
[2019-02-06 14:26:09.258000] [DEBUG] command with ID VA1V finished with return code 0
[2019-02-06 14:26:09.263704] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendering with Jinja2...
[2019-02-06 14:26:09.264322] [DEBUG] br0ken:br0ken:k8s_deployment:br0ken/nginx: rendered in 0.000183s
[2019-02-06 14:26:09.273939] [DEBUG] worker pool apply_br0ken delivering result of br0ken:k8s_deployment:br0ken/nginx on worker #3
[2019-02-06 14:26:09.274299] [DEBUG] skipped these items because they depend on k8s_deployment:br0ken/nginx, which was skipped previously: bundle:br0ken, k8s_deployment:
[2019-02-06 14:26:09.274438] [DEBUG] skipped these items because they depend on bundle:br0ken, which was skipped previously: bundle:
[2019-02-06 14:26:09.274657] ✘ br0ken  br0ken  k8s_deployment:br0ken/nginx failed (manifest)
[2019-02-06 14:26:09.274757] [DEBUG] shutting down worker pool apply_br0ken
[2019-02-06 14:26:09.277190] [DEBUG] worker pool apply_br0ken has been shut down
[2019-02-06 14:26:09.277454] i br0ken  completed after 1s  (2 OK, 0 fixed, 0 skipped, 1 failed)
[2019-02-06 14:26:09.277743] [DEBUG] worker pool apply delivering result of br0ken on worker #0
[2019-02-06 14:26:09.277922] [DEBUG] shutting down worker pool apply
[2019-02-06 14:26:09.278096] [DEBUG] worker pool apply has been shut down
[2019-02-06 14:26:09.278394] i ╭────────┬───────┬────┬───────┬─────────┬────────┬──────╮
[2019-02-06 14:26:09.278557] i │ node   │ items │ OK │ fixed │ skipped │ failed │ time │
[2019-02-06 14:26:09.278666] i ├────────┼───────┼────┼───────┼─────────┼────────┼──────┤
[2019-02-06 14:26:09.278811] i │ br0ken │     3 │  2 │     0 │       0 │      1 │   1s │
[2019-02-06 14:26:09.278894] i ╰────────┴───────┴────┴───────┴─────────┴────────┴──────╯
```

**Comment by trehn** [comment]
fixed in #516
